doc/builders/trivial-builders.chapter.md

```diff
@@ -0,0 +1,52 @@
+# Trivial builders {#chap-trivial-builders}
+
+Nixpkgs provides a couple of functions that help with building derivations. The most important one, `stdenv.mkDerivation`, has already been documented above. The following functions wrap `stdenv.mkDerivation`, making it easier to use in certain cases.
+
+## `runCommand`{#trivial-builder-runCommand}
+
+This takes three arguments, `name`, `env`, and `buildCommand`. `name` is just the name that Nix will append to the store path in the same way that `stdenv.mkDerivation` uses its `name` attribute. `env` is an attribute set specifying environment variables that will be set for this derivation. These attributes are then passed to the wrapped `stdenv.mkDerivation`. `buildCommand` specifies the commands that will be run to create this derivation. Note that you will need to create `$out` for Nix to register the command as successful.
+
+An example of using `runCommand` is provided below.
+
+```nix
+(import <nixpkgs> {}).runCommand "my-example" {} ''
+  echo My example command is running
+
+  mkdir $out
+
+  echo I can write data to the Nix store > $out/message
+
+  echo I can also run basic commands like:
+
+  echo ls
+  ls
+
+  echo whoami
+  whoami
+
+  echo date
+  date
+''
+```
+
+## `runCommandCC` {#trivial-builder-runCommandCC}
+
+This works just like `runCommand`. The only difference is that it also provides a C compiler in `buildCommand`'s environment. To minimize your dependencies, you should only use this if you are sure you will need a C compiler as part of running your command.
+
+## `runCommandLocal` {#trivial-builder-runCommandLocal}
+
+Variant of `runCommand` that forces the derivation to be built locally, it is not substituted. This is intended for very cheap commands (<1s execution time). It saves on the network roundrip and can speed up a build.
+
+    ::: {#special .note}
```

Comment:
Also why are you adding `special` anchor?

Reply:
I just followed the tutorial.

Reply:
In Markdown extensions, attribute `#foo` is synonymous to `id=foo` and `.bar` means `class=bar` (probably originates from CSS).

For `note` admonition, you will want `note` class so that is 

```markdown
::: {.note}
foo
:::
```

When there is only a single class, pandoc allows to use shorter

```markdown
::: note
foo
:::
```

That is equivalent and probably preferred.